File: Sources/1010/1010.c
/**
 *  BOJ 1010번 C언어 소스 코드
 *  작성자 : 동동매니저 (DDManager)
 *
 *  ※ 실행 결과
 *  사용 메모리 : 1,116 KB / 131,072 KB
 *  소요 시간 : 0 ms / 2,000 ms
 *
 *  Copyright 2019. DDManager all rights reserved.
 */

#include <stdio.h>

void C(int,int);

int main(void) {
	int T,N,M,i;
	scanf("%d",&T);
	for(i=0;i<T;i++){
		scanf("%d %d",&N,&M);
		C(N,M);
	}
	return 0;
}

void C(int N,int M) {
	int v=1,l;
	if(M-N<N) N=M-N;
	for(l=1;l<=N;l++){
		v*=M-l+1;
		v/=l;
	}
	printf("%d\n",v);
}

File: Sources/1010/1010.java
/**
 *  BOJ 1010번 Java 소스 코드
 *  작성자 : 동동매니저 (DDManager)
 *
 *  ※ 실행 결과
 *  사용 메모리 : 15,896 KB / 278,528 KB
 *  소요 시간 : 172 ms / 5,000 ms
 *
 *  Copyright 2019. DDManager all rights reserved.
 */

import java.util.Scanner;
public class Main{
	public static void main(String[] args){
		Scanner sc=new Scanner(System.in);
		int T=sc.nextInt();
		for(int i=0;i<T;i++) System.out.println(C(sc.nextInt(),sc.nextInt()));
		sc.close();
	}
	private static int C(int N,int M){
		int v=1;
		if(M-N<N) N=M-N;
		for(int i=1;i<=N;i++){
			v*=M-i+1;
			v/=i;
		}
		return v;
	}
}